### PR TITLE
9554 integrity2

### PR DESCRIPTION
**Pull request by cortlandstarrett**
Check Referential Integrity ...pending build with unit tests

### CODE REVIEW

File: src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            String projPath = project.getLocation().toOSString();
+            IPath modelspath = new Path(projPath + File.separator + "models");
+            if (modelspath.toFile().exists()) {
+                modelsDir.add("-i");
+                modelsDir.add(modelspath.toOSString());
+                if ( "" == destPath ) {
+                    // only set these up on the first project found
+                    path = new Path(projPath + File.separator + DOC_DIR + File.separator);
+                    if (!path.toFile().exists()) {
+                        path.toFile().mkdir();
```
Comment (keithbrown) [inline src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java]: Let's use mkdirs() instead of mkdir()

File: doc-bridgepoint/notes/9567_association_editing_enhancements/cme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+5.1.2.1 Another CME Approach
```
Comment (keithbrown) [inline doc-bridgepoint/notes/9567_association_editing_enhancements/cme.md]: Why is this file here?  Seems like we need to remerge master.

File: src/org.xtuml.bp.integrity/.externalToolBuilders/Build.launch [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<launchConfiguration type="org.eclipse.ui.externaltools.ProgramBuilderLaunchConfigurationType">
+<stringAttribute key="org.eclipse.debug.core.ATTR_REFRESH_SCOPE" value="${workspace}"/>
+<mapAttribute key="org.eclipse.debug.core.environmentVariables">
+<mapEntry key="INCLUDE_TESTS" value="true"/>
+<mapEntry key="bp_install_dir" value="${eclipse_home}"/>
+</mapAttribute>
+<booleanAttribute key="org.eclipse.debug.ui.ATTR_LAUNCH_IN_BACKGROUND" value="false"/>
+<stringAttribute key="org.eclipse.ui.externaltools.ATTR_LOCATION" value="${system_path:mvn}"/>
+<stringAttribute key="org.eclipse.ui.externaltools.ATTR_RUN_BUILD_KINDS" value="full,incremental,auto,"/>
+<stringAttribute key="org.eclipse.ui.externaltools.ATTR_TOOL_ARGUMENTS" value="-Dtycho.disableP2Mirrors=true -Dmaven.test.failure.ignore=true -DskipTests=true -f ${git_dir:org.xtuml.bp.releng.parent}/../releng/org.xtuml.bp.releng.parent/pom.xml -pl org.xtuml.bp:org.xtuml.bp.docgen install"/>
```
Comment (keithbrown) [inline src/org.xtuml.bp.integrity/.externalToolBuilders/Build.launch]: I see org.xtuml.bp.docgen in this line.  I'm guessing that should be org.xtuml.bp.integrity.  This makes me thing we should do a search inside this plugin for "bp.docgen" and see if there are any other places.

File: src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                        while (curStep <= steps) {
+                            if (monitor.isCanceled()) {
+                                InterruptedException ie = new InterruptedException("User cancelled the operation");
+                                throw ie;
+                            }
+
+                            try {
+                                switch (curStep) {
+                                case 1:
+                                    monitor.subTask("Accumulating model data for workspace");
```
Comment (keithbrown) [inline src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java]: This step is pointless.  It doesn't do any work.  The user will never see the message because it will go by so fast.

File: doc-bridgepoint/notes/9554_integrity_int.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+exclude this data so that Prebuilder output can be used as integrity input.
+
+6.5 Out-and-Back  
+A select out-and-back statement was added to detect when some associations
+may have been over-written with batch relate.  This covers the case of
+an extra subtype that the instance loader and batch relater would allow
+to occur.
+
+6.6 `defaults.xtuml`  
+This file was include in bp.core and is no longer used.  Deleting it.
```
Comment (keithbrown) [inline doc-bridgepoint/notes/9554_integrity_int.md]: include > included

File: doc-bridgepoint/notes/9554_integrity_int.md [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+The plugin depends on io.image and `System_c` (`S_SYS`).  These dependencies
+are only in place because needing to have the _project_ variable.  This
+can be obtained in simpler ways.
+
+6.9 bp.core ooaofooa  
+See [[2.5]](#2.5).  
+
+6.10 Enhanced Loader  
+Consider fixing the loader to deal with unrecognized data gracefully.  
+Consider enhancing the escher loader to support OOA of Graphics instances.
```
Comment (keithbrown) [inline doc-bridgepoint/notes/9554_integrity_int.md]: "Consider?"  So was work performed, or follow-on issues raised?

File: src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java (hunk 4 above)
Comment (cortlandstarrett) [inline src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java]: It makes a big difference.  :)  It means that the progress bar shows a fraction of progress while they wait for the real work rather than nothing and then all.

File: doc-bridgepoint/notes/9567_association_editing_enhancements/cme.md (hunk 2 above)
Comment (cortlandstarrett) [inline doc-bridgepoint/notes/9567_association_editing_enhancements/cme.md]: This is actually my contribution to the note.

File: src/org.xtuml.bp.integrity/.externalToolBuilders/Build.launch (hunk 3 above)
Comment (cortlandstarrett) [inline src/org.xtuml.bp.integrity/.externalToolBuilders/Build.launch]: I will edit this in place.

File: doc-bridgepoint/notes/9554_integrity_int.md (hunk 6 above)
Comment (cortlandstarrett) [inline doc-bridgepoint/notes/9554_integrity_int.md]: Made this part of the follow-on issue...
https://support.onefact.net/issues/9742
